src/objects/zcl_abapgit_object_odso.clas.abap

```diff
@@ -0,0 +1,256 @@
+CLASS zcl_abapgit_object_odso DEFINITION
+  PUBLIC
+  INHERITING FROM zcl_abapgit_objects_super
+  FINAL
+  CREATE PUBLIC .
+
+  PUBLIC SECTION.
+
+    INTERFACES zif_abapgit_object .
+
+    ALIASES mo_files
+      FOR zif_abapgit_object~mo_files .
+  PROTECTED SECTION.
+  PRIVATE SECTION.
+ENDCLASS.
+
+
+CLASS zcl_abapgit_object_odso IMPLEMENTATION.
+
+
+  METHOD zif_abapgit_object~changed_by.
+
+    DATA: lv_dsonam  TYPE rsobjvers,
+          ls_return  TYPE bapiret2,
+          ls_details TYPE bapi6116.
+
+    lv_dsonam = ms_item-obj_name.
+
+    CALL FUNCTION 'BAPI_ODSO_GETDETAIL'
+      EXPORTING
+        odsobject = lv_dsonam
+      IMPORTING
+        details   = ls_details
+        return    = ls_return.
+
+    IF ls_return-type = 'E'.
+      zcx_abapgit_exception=>raise( |Error when geting changed by of ODSO: { ls_return-message }| ).
+    ENDIF.
+
+    rv_user = ls_details-tstpnm.
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~delete.
+
+    DATA: lv_odso       TYPE rsdodsobject,
+          lv_objname    TYPE sobj_name,
+          lr_collection TYPE REF TO cl_rsd_odso_collection,
+          lt_msg        TYPE rs_t_msg,
+          ls_msg        TYPE bal_s_msg.
+
+    lv_odso = ms_item-obj_name.
+    lv_objname =  ms_item-obj_name.
+
+    CREATE OBJECT lr_collection.
+
+    TRY.
+        lr_collection->add_tlogo(
+            i_objnm          = lv_objname
+            i_modify         = abap_true
+            i_delete         = abap_true  ).
+
+        lr_collection->delete( ).
+
+        cl_rso_application_log=>appl_log_msg_read(
+          IMPORTING
+            e_t_msg             =  lt_msg ).
+
+        READ TABLE lt_msg WITH KEY msgty = 'E' INTO ls_msg.
+        IF sy-subrc = 0.
+          zcx_abapgit_exception=>raise(
+          |Error when creating ODSO: { ms_item-obj_name } { ls_msg-msgv1 } { ls_msg-msgv2 }| ).
+        ENDIF.
+
+      CATCH cx_rs_cancelled.
+        zcx_abapgit_exception=>raise( |Canceled deletion of ODSO: { ms_item-obj_name }| ).
+      CATCH cx_rs_existing.
+      CATCH cx_rs_not_found.
+        zcx_abapgit_exception=>raise( |ODSO not found: { ms_item-obj_name }| ).
+    ENDTRY.
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~deserialize.
+
+    DATA: lv_odso        TYPE rsdodsobject,
+          ls_details     TYPE bapi6116,
+          lt_infoobjects TYPE STANDARD TABLE OF bapi6116io,
+          lt_navigation  TYPE STANDARD TABLE OF bapi6116na,
+          lt_indexes     TYPE STANDARD TABLE OF bapi6116in,
+          lt_index_iobj  TYPE STANDARD TABLE OF bapi6116ii,
+          lt_return      TYPE STANDARD TABLE OF bapiret2,
+          ls_return      TYPE bapiret2.
+
+    IF  cl_rs_b4hana_util=>get_b4hana_switch( ) = cl_rs_b4hana_util=>n_c_switch-b4hana.
```

Comment:
Suggest to change to dynamic call, as the class is not available in lower release.

Reply:
and no need to do this check, run the code and fail if it fails instead